nimterop/getters.nim

```diff
@@ -695,3 +694,16 @@ proc expandSymlinkAbs*(path: string): string =
     result = path.expandSymlink().absolutePath(path.parentDir()).normalizedPath()
   except:
     result = path
+
+proc determineCompilerMode(path: string) :string =
+  let file = path.splitFile()
+  if file.ext in [".hxx", ".hpp", ".hh", ".H", ".h++", ".cpp", ".cxx", ".cc", ".C", ".c++"]:
+    result = "cpp"
+  else:
+    result = "c"
```

Comment:
So we are defaulting to C mode if not C++ files? No more complaining if unsupported filetype?

Reply:
I think that makes the most sense, considering the test in PCRE: https://github.com/nimterop/nimterop/blob/master/tests/tpcre.nim

Reply:
Either that or that test should be changed, and maybe we could raise an exception for unsupported filetypes

Reply:
In that case, tpcre.nim should specify the `-mc` flag in cImport so that the mode is explicit. It might break some wrappers out there but I think it is better to be explicit.